bliss/encoder/variational_dist.py

```diff
@@ -79,9 +76,18 @@ def compute_nll(self, params, true_tile_cat):
             gating = rearrange(true_tile_cat.star_bools, "b ht wt 1 1 -> b ht wt")
         elif self.nll_gating == "is_galaxy":
             gating = rearrange(true_tile_cat.galaxy_bools, "b ht wt 1 1 -> b ht wt")
+        elif self.nll_gating == "is_in_cosmodc2":
+            gating = rearrange(true_tile_cat["cosmodc2_mask"], "b ht wt 1 1 -> b ht wt")
```

Comment:
Could you made this change in a subclass of `VariationalDist` instead?

Reply:
No problem.

Reply:
Write some gating classes to realize this